### PR TITLE
General maintenance for RPED

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
 **1. Code Improvements**
 - Converted vars for playing RPED sounds into defines & proc overrides. saved some memory
 - Created separate storage type for bluespace RPED(cyborg variant had the same values so it shares this type)
 - Improved its datum storage code
 - Moved RPED into its own file out of stock parts. Better file organization
 - Autodoc for some procs

**2. Refactor**
 - Strictly separates bluespace RPED interaction from normal one. What this means is
   - Removed var `works_from_distance` we now typecheck for bluespace RPED specifically
   - Moved `ranged_interact_with_atom()` from normal RPED into its bluespace subtype. We now play the beam animation from inside this proc rather than redundantly checking for its bluespace variant inside machines `exchange_parts()` & playing it there
   - No more snowflacky check for frames. That's handled inside its `ranged_item_interaction()` proc

Just overall cleaner code

## Changelog
:cl:
code: improved code for RPED
refactor: RPED attack chain has been refactored. Reports bugs on github
/:cl:

### CODE REVIEW

File: code/modules/research/stock_parts.dm [hunk 1]
```diff
@@ -10,72 +14,76 @@ If you create T5+ please take a pass at mech_fabricator.dm. The parts being good
...
+	. = ..()
+	if(. & ITEM_INTERACT_ANY_BLOCKER)
+		user.Beam(attacked_object, icon_state = "rped_upgrade", time = 0.5 SECONDS)
+
+/obj/item/storage/part_replacer/bluespace/ranged_interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	return interact_with_atom(interacting_with, user, modifiers)
+
+/obj/item/storage/part_replacer/bluespace/play_rped_sound()
+	if(prob(1))
+		playsound(src, USE_SOUND, 40, TRUE)
```
Comment (Sadboysuss) [inline code/modules/research/stock_parts.dm:83]: I think this is meant to be pshoom2

File: code/modules/research/stock_parts.dm [hunk 2]
```diff
@@ -10,72 +14,76 @@ If you create T5+ please take a pass at mech_fabricator.dm. The parts being good
...
+	else
```
Comment (Sadboysuss) [inline code/modules/research/stock_parts.dm:84]: Could do early returns here instead?

Comment (SyncIt21) [reply code/modules/research/stock_parts.dm:84]: You right.

File: code/modules/research/stock_parts.dm (hunk 1 above)
Comment (SyncIt21) [reply code/modules/research/stock_parts.dm:83]: You right. I'll swap that

File: code/modules/research/stock_parts.dm [hunk 3]
```diff
@@ -10,72 +14,76 @@ If you create T5+ please take a pass at mech_fabricator.dm. The parts being good
...
+		user.Beam(attacked_object, icon_state = "rped_upgrade", time = 0.5 SECONDS)
+
+/obj/item/storage/part_replacer/bluespace/ranged_interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	return interact_with_atom(interacting_with, user, modifiers)
+
+/obj/item/storage/part_replacer/bluespace/play_rped_sound()
+	if(prob(1))
+		playsound(src, 'sound/items/pshoom/pshoom.ogg', 40, TRUE)
+		return
+	playsound(src, USE_SOUND, 40, TRUE)
```
Comment (Sadboysuss) [inline code/modules/research/stock_parts.dm:85]: isn't this gonna play the normal rped sound? You'd want pshoom.ogg here

File: code/modules/research/stock_parts.dm [hunk 4]
```diff
@@ -10,72 +14,76 @@ If you create T5+ please take a pass at mech_fabricator.dm. The parts being good
...
+	. = ..()
+	if(. & ITEM_INTERACT_ANY_BLOCKER)
+		user.Beam(attacked_object, icon_state = "rped_upgrade", time = 0.5 SECONDS)
+
+/obj/item/storage/part_replacer/bluespace/ranged_interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	return interact_with_atom(interacting_with, user, modifiers)
+
+/obj/item/storage/part_replacer/bluespace/play_rped_sound()
+	if(prob(1))
+		playsound(src, 'sound/items/pshoom/pshoom.ogg', 40, TRUE)
```
Comment (Sadboysuss) [inline code/modules/research/stock_parts.dm:83]: the previous behavior for BRPEDs played pshoom2.ogg with a probability of 1

File: code/datums/storage/subtypes/rped.dm [hunk 5]
```diff
@@ -33,11 +32,22 @@
 	)
 
 /datum/storage/rped/can_insert(obj/item/to_insert, mob/user, messages = TRUE, force = FALSE)
-	. = ..()
-	if(!.)
-		return .
+	//only stock parts permited
+	if(!to_insert.get_part_rating())
```
Comment (Ghommie) [inline code/datums/storage/subtypes/rped.dm:36]: May I suggest an early return even if that means adding an extra `return ..()`

**Review by Ghommie — APPROVED** [review]

**Pull request merged**